### PR TITLE
Part one of engine deprecation

### PR DESCRIPTION
**Pull request by DanG100**
Maintaining both engine and saiserver API is becoming difficult. 

First step of deprecation making the saiserver not rely on engine.
Next step will migrating the gnmi reconcilers to use the sai APIs

Review note: for now this is introducing a lot of code duplication, but this is short term problem

### CODE REVIEW

File: dataplane/internal/engine/engine.go [hunk 1]
```diff
@@ -78,10 +77,11 @@ type Engine struct {
 	// internalToExternalID is a map from the internal port id to it's corresponding external port.
 	internalToExternalID map[string]string
 	cancelFn             func()
+	createTables         bool
 }
 
 // New creates a new engine and sets up the forwarding tables.
-func New(ctx context.Context) (*Engine, error) {
+func New(ctx context.Context, createTables bool) (*Engine, error) {
```
Comment (wenovus) [inline dataplane/internal/engine/engine.go:84]: Add comment for why you might not want to create tables.

File: dataplane/standalone/proto/common.proto
Comment (wenovus) [inline dataplane/standalone/proto/common.proto:1]: Curious what's with the number changes?

**Review by wenovus — APPROVED** [review]

File: dataplane/internal/engine/engine.go (hunk 1 above)
Comment (DanG100) [reply dataplane/internal/engine/engine.go]: added

File: dataplane/standalone/proto/common.proto
Comment (DanG100) [reply dataplane/standalone/proto/common.proto:1]: the c++ to proto conversion code does enum conversion as raw int assignement so the numbers need to match up correctly

**Pull request merged**